whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java

```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
 
     @Nullable
     public EmailMessageInfo parse(final MimeMessage message) throws MessagingException, MailParsingException {
-        if (!enabled || !isMultipartReport(message) ){
+        if (!enabled) {
             return null;
         }
 
-        try {
-            final MultipartReport multipartReport = multipartReport(message.getContent());
-            if (isReportDeliveryStatus(multipartReport)) {
-                final DeliveryStatus deliveryStatus = deliveryStatus(message);
-                if (isFailed(deliveryStatus)) {
-                    final MimeMessage returnedMessage = multipartReport.getReturnedMessage();
-                    final String messageId = getMessageId(returnedMessage.getMessageID());
-                    final List<String> recipient = extractRecipients(deliveryStatus);
-                    return new EmailMessageInfo(recipient, messageId);
+        if (isMultipartReport(message)) {
+            try {
+                final MultipartReport multipartReport = multipartReport(message.getContent());
+                if (isReportDeliveryStatus(multipartReport)) {
```

Comment:
Line 58 - 66 and 77 -85 are the same
can we extract this into a method called something like "extractMailMessageInfo"?

Reply:
I don't want to disturb the existing code, and they are a bit different still - we throw an exception in the multipart/report code that we don't throw in the multipart/mixed code.
I added a TODO for now.